presto-hive/src/main/java/io/prestosql/plugin/hive/DynamicConfigurationProvider.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.hive;
+
+import org.apache.hadoop.conf.Configuration;
+
+import java.net.URI;
+
+import static io.prestosql.plugin.hive.HdfsEnvironment.HdfsContext;
+
+public interface DynamicConfigurationProvider
+{
+    Configuration updateConfiguration(Configuration configuration, HdfsContext context, URI uri);
```

Comment:
Please remove the return type. It isn't used and the semantics of it are not clear (since the method is supposed to update the passed in configuration object).

Reply:
This was fixed, but the change is in the `Rename` commit by mistake. I moved it to the commit where I introduce `DynamicConfigurationProvider`.